### ISSUE TITLE
[Bug] Create Data for Waymo stopped mid-way

### ISSUE THREAD

**Issue by anhquan3012** [open-mmlab/mmdetection3d#2295]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection3d/issues) and [Discussions](https://github.com/open-mmlab/mmdetection3d/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://github.com/open-mmlab/mmdetection3d/blob/1.1/docs/en/notes/faq.md) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (dev)](https://github.com/open-mmlab/mmdetection3d/tree/dev) or [latest version (1.x)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmdetection3d

### Environment

sys.platform: linux
Python: 3.8.16 (default, Jan 17 2023, 23:13:24) [GCC 11.2.0]
CUDA available: True
GPU 0,1,2,3: GeForce RTX 3090
CUDA_HOME: /usr/local/cuda-11.3
NVCC: Cuda compilation tools, release 11.3, V11.3.58
GCC: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
PyTorch: 1.7.1
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.8.0a0
OpenCV: 4.7.0
MMCV: 1.6.2
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 11.0
MMDetection: 2.28.1
MMSegmentation: 0.30.0
MMDetection3D: 1.0.0rc6+5f7a85d
spconv2.0: False

----------PIP packages----------
Package                     Version             Editable project location
--------------------------- ------------------- -----------------------------------------------------------
absl-py                     0.15.0
addict                      2.4.0
anyio                       3.6.2
argon2-cffi                 21.3.0
argon2-cffi-bindings        21.2.0
arrow                       1.2.3
asttokens                   2.2.1
astunparse                  1.6.3
attrs                       22.2.0
backcall                    0.2.0
beautifulsoup4              4.11.2
black                       23.1.0
bleach                      6.0.0
brotlipy                    0.7.0
cachetools                  5.3.0
certifi                     2022.12.7
cffi                        1.15.1
charset-normalizer          2.0.4
click                       8.1.3
colorama                    0.4.6
comm                        0.1.2
ConfigArgParse              1.5.3
contourpy                   1.0.7
cryptography                38.0.4
cycler                      0.11.0
dash                        2.8.1
dash-core-components        2.0.0
dash-html-components        2.0.0
dash-table                  5.0.0
debugpy                     1.6.6
decorator                   5.1.1
defusedxml                  0.7.1
descartes                   1.1.0
exceptiongroup              1.1.0
executing                   1.2.0
fastjsonschema              2.16.2
fire                        0.5.0
flake8                      6.0.0
Flask                       2.2.2
flatbuffers                 1.12
flit_core                   3.6.0
fonttools                   4.38.0
fqdn                        1.5.1
gast                        0.4.0
google-auth                 2.16.0
google-auth-oauthlib        0.4.6
google-pasta                0.2.0
grpcio                      1.34.1
h5py                        3.1.0
idna                        3.4
imageio                     2.25.0
importlib-metadata          6.0.0
importlib-resources         5.10.2
iniconfig                   2.0.0
ipykernel                   6.21.1
ipython                     8.10.0
ipython-genutils            0.2.0
ipywidgets                  8.0.4
isoduration                 20.11.0
itsdangerous                2.1.2
jedi                        0.18.2
Jinja2                      3.1.2
joblib                      1.2.0
jsonpointer                 2.3
jsonschema                  4.17.3
jupyter                     1.0.0
jupyter_client              8.0.2
jupyter-console             6.5.0
jupyter_core                5.2.0
jupyter-events              0.6.3
jupyter_server              2.2.1
jupyter_server_terminals    0.4.4
jupyterlab-pygments         0.2.2
jupyterlab-widgets          3.0.5
keras-nightly               2.5.0.dev2021032900
Keras-Preprocessing         1.1.2
kiwisolver                  1.4.4
llvmlite                    0.36.0
lyft-dataset-sdk            0.0.8
Markdown                    3.4.1
markdown-it-py              2.1.0
MarkupSafe                  2.1.2
matplotlib                  3.6.3
matplotlib-inline           0.1.6
mccabe                      0.7.0
mdurl                       0.1.2
mistune                     2.0.5
mkl-fft                     1.3.1
mkl-random                  1.2.2
mkl-service                 2.4.0
mmcls                       0.25.0
mmcv-full                   1.6.2
mmdet                       2.28.1
mmdet3d                     1.0.0rc6            /media/ntu/volume1/home/s122md304_13/quan_fyp/mmdetection3d
mmsegmentation              0.30.0
model-index                 0.1.11
mypy-extensions             1.0.0
nbclassic                   0.5.1
nbclient                    0.7.2
nbconvert                   7.2.9
nbformat                    5.5.0
nest-asyncio                1.5.6
networkx                    2.2
notebook                    6.5.2
notebook_shim               0.2.2
numba                       0.53.0
numpy                       1.19.5
nuscenes-devkit             1.1.9
oauthlib                    3.2.2
open3d                      0.16.0
opencv-python               4.7.0.68
openmim                     0.3.6
opt-einsum                  3.3.0
ordered-set                 4.1.0
packaging                   23.0
pandas                      1.4.3
pandocfilters               1.5.0
parso                       0.8.3
pathspec                    0.11.0
pexpect                     4.8.0
pickleshare                 0.7.5
Pillow                      9.3.0
pip                         22.3.1
pkgutil_resolve_name        1.3.10
platformdirs                3.0.0
plotly                      5.13.0
pluggy                      1.0.0
plyfile                     0.7.4
prettytable                 3.6.0
prometheus-client           0.16.0
prompt-toolkit              3.0.36
protobuf                    3.9.2
psutil                      5.9.4
ptyprocess                  0.7.0
pure-eval                   0.2.2
pyasn1                      0.4.8
pyasn1-modules              0.2.8
pycocotools                 2.0.6
pycodestyle                 2.10.0
pycparser                   2.21
pyflakes                    3.0.1
Pygments                    2.14.0
pyOpenSSL                   22.0.0
pyparsing                   3.0.9
pyquaternion                0.9.9
pyrsistent                  0.19.3
PySocks                     1.7.1
pytest                      7.2.1
python-dateutil             2.8.2
python-json-logger          2.0.4
pytz                        2022.7.1
PyWavelets                  1.4.1
PyYAML                      6.0
pyzmq                       25.0.0
qtconsole                   5.4.0
QtPy                        2.3.0
requests                    2.28.1
requests-oauthlib           1.3.1
rfc3339-validator           0.1.4
rfc3986-validator           0.1.1
rich                        13.3.1
rsa                         4.9
scikit-image                0.19.3
scikit-learn                1.2.1
scipy                       1.10.0
seaborn                     0.12.2
Send2Trash                  1.8.0
setuptools                  65.6.3
shapely                     2.0.1
six                         1.15.0
sniffio                     1.3.0
soupsieve                   2.3.2.post1
stack-data                  0.6.2
tabulate                    0.9.0
tenacity                    8.2.1
tensorboard                 2.11.2
tensorboard-data-server     0.6.1
tensorboard-plugin-wit      1.8.1
tensorflow                  2.5.0
tensorflow-estimator        2.5.0
termcolor                   1.1.0
terminado                   0.17.1
terminaltables              3.1.10
threadpoolctl               3.1.0
tifffile                    2023.2.3
tinycss2                    1.2.1
tomli                       2.0.1
torch                       1.7.1
torchaudio                  0.7.0a0+a853dff
torchvision                 0.8.0a0
tornado                     6.2
tqdm                        4.64.1
traitlets                   5.9.0
trimesh                     2.35.39
typing-extensions           3.7.4.3
uri-template                1.2.0
urllib3                     1.26.14
waymo-open-dataset-tf-2-5-0 1.4.1
wcwidth                     0.2.6
webcolors                   1.12
webencodings                0.5.1
websocket-client            1.5.1
Werkzeug                    2.2.2
wheel                       0.37.1
widgetsnbextension          4.0.5
wrapt                       1.12.1
yapf                        0.32.0
zipp                        3.13.0

### Reproduces the problem - code sample

NA

### Reproduces the problem - command or script

```
python tools/create_data.py waymo --root-path /media/ntu/volume1/home/s122md304_13/quan_fyp/waymo --out-dir /media/ntu/volume1/home/s122md304_13/quan_fyp/waymo --workers 128 --extra-tag waymo
```

### Reproduces the problem - error message

There was no error message. However, the data generation stopped after 197 samples, although the process was still running. The progress said ETA is 200000+ seconds, while the tasks/s said 0.0 tasks/s.

### Additional information

I think it might be because I did not use the correct Waymo dataset version? I downloaded version 1.2 as instructed in the documentation. However, the version of waymo-open-dataset-tf I used is waymo-open-dataset-tf-2-5-0, as I cannot install waymo-open-dataset-tf-2-1-0==1.2.0 as instructed.

**Comment by JingweiZhang12** [comment]
@anhquan3012 I suggest you can use a small number of workers, such as --workers=32

**Comment by anhquan3012** [comment]
Hi. I did decrease the number of workers to 16, but it did not solve the problem. The data generation always stopped at this point (sample 197 for each interval).
![image](https://user-images.githubusercontent.com/108224462/220376884-0cea3ed3-6b6c-49b5-8349-7c82c274ad23.png)
![image](https://user-images.githubusercontent.com/108224462/220376804-4a5ee045-ef74-487f-9705-4619a06191b2.png)
The log kept looking like this and I saw no further progress.
![image](https://user-images.githubusercontent.com/108224462/220377103-8e7549d8-369a-450d-8d94-579299052248.png)

The Waymo dataset version that I used is 1.2. Should that be the correct dataset version?

**Comment by JingweiZhang12** [comment]
The version of Waymo has no problem. Suggest you could close the multiprocess and profile this problem.

**Comment by Xiangxu-0103** [comment]
We name the filename as `{a}{bbb}{ccc}`, where `a` is the prefix of different split (training/validation/testing), `bbb` for segment index and `ccc` for frame index. For `000` segment index, it has 198 frames, so it will generate filenames from `0000000` to `0000197`, without `0000198` and so on. So everything is OK for me.

**Comment by anhquan3012** [comment]
Ok thank you very much. I realized that the processing was not stopping, just generating data very slow. There was no issue here.
Btw, if I want to use only part of the Waymo training data, can I just modify the file ImageSets/train.txt to contain only the samples that I want to use?

**Comment by Xiangxu-0103** [comment]
> Ok thank you very much. I realized that the processing was not stopping, just generating data very slow. There was no issue here. Btw, if I want to use only part of the Waymo training data, can I just modify the file ImageSets/train.txt to contain only the samples that I want to use?

Yes, you can modify the file to generate part of training set.
